src/explore-education-statistics-admin/src/pages/release/content/utils/getUnresolvedComments.ts

```diff
@@ -1,38 +1,49 @@
 import { EditableRelease } from '@admin/services/releaseContentService';
-import { Comment, EditableBlock } from '@admin/services/types/content';
-import { ContentSection } from '@common/services/publicationService';
+import { EditableBlock } from '@admin/services/types/content';
+import { BlockCommentIds } from '@admin/contexts/EditingContext';
+
+export const getTotalUnresolvedComments = (
+  unresolvedComments: BlockCommentIds,
+  unsavedCommentDeletions: BlockCommentIds,
+) => {
+  const allUnresolved = Object.values(unresolvedComments).flat();
+  const allUnsaved = Object.values(unsavedCommentDeletions).flat();
+  return allUnresolved.filter(id => !allUnsaved.includes(id)).length;
+};
 
 const getContentSectionComments = (
-  contentSection: ContentSection<EditableBlock>,
-): Comment[] => {
-  if (contentSection.content?.length) {
-    return contentSection.content.reduce<Comment[]>(
-      (allCommentsForSection, content) => {
-        content.comments.forEach(comment =>
-          allCommentsForSection.push(comment),
-        );
-        return allCommentsForSection;
-      },
-      [],
-    );
+  blocks: EditableBlock[],
+): BlockCommentIds => {
+  if (!blocks || !blocks.length) {
+    return {};
   }
+  const unresolvedComments: BlockCommentIds = {};
 
-  return [];
+  blocks.forEach(block => {
+    if (!unresolvedComments[block.id]) {
+      unresolvedComments[block.id] = [];
+    }
+    block.comments.forEach(comment => {
+      if (!comment.resolved) {
+        unresolvedComments[block.id].push(comment.id);
+      }
+    });
+  });
+  return unresolvedComments;
```

Comment:
Could refactor this to use `reduce` to make it a little more succinct:

```ts
return blocks.reduce<BlockCommentIds>((acc, block) => {
  if (!acc[block.id]) {
    acc[block.id] = [];
  }
  
  block.comments.forEach(comment => {
    if (!comment.unresolved) {
      acc[block.id].push(comment.id);
    }
  });

  return acc;
}, {});
```

Reply:
Done